### PR TITLE
Update to E2E 1.xx

### PR DESCRIPTION
**Pull request by Discomanco**
First round of fixes.
So far, Cosmic Meatballs seems to potentially only happen on upgraded clients, needs further testing.
Dark Steel Upgrades is more complicated than anticipated
Unbreakable IE Rods is user error, not sure if fixable

### CODE REVIEW

**Comment by ReiDaTecnologia** [comment]
Rip Animania you will be slightly missed xD

**Comment by Discomanco** [comment]
-Animania
-Extreme Reactors
+Endergy
Scripts from E2E-V2 used, still needs testing and polishing

**Comment by ReiDaTecnologia** [comment]
Rip Extreme Reactors, you will not be missed xD

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,7 @@ config/AppliedEnergistics2/VersionChecker.cfg
 config/astralsorcery/amulet_enchantments.cfg
 config/brandon3055/contributors.json
 config/brandon3055/Custom Fusion Recipe Info.txt
+config/vanillafix.cfg
+config/vanillafix/config_version
+config/vanillafix/methods-stable_39.csv
+config/splash.properties
```
Comment (NielsPilgaard) [inline .gitignore:108]: we need that in I'm afraid, it configures the startup screen.

File: scripts/DraconicEvolution.zs [hunk 2]
```diff
@@ -11,7 +11,7 @@ import mods.jei.JEI.removeAndHide as rh;
 	recipes.addShapedMirrored("Mob Grinder", 
 	<draconicevolution:grinder>, 
 	[[<ore:ingotElectricalSteel>, <draconicevolution:draconium_block:1>, <ore:ingotElectricalSteel>],
-	[<astralsorcery:itemcrystalsword>.anyDamage(), <draconicevolution:draconic_core>, <astralsorcery:itemcrystalsword>.anyDamage()], 
+	[<astralsorcery:itemcrystalsword>.onlyDamageAtLeast(0), <draconicevolution:draconic_core>, <astralsorcery:itemcrystalsword>.onlyDamageAtLeast(0)],
```
Comment (NielsPilgaard) [inline scripts/DraconicEvolution.zs:14]: Does this work? :D

File: scripts/Endergy.zs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	<astralsorcery:itemcraftingcomponent:2>, #<astralsorcery:itemshiftingstar>.withTag({}), 
+	<astralsorcery:itemcraftingcomponent:4>, <astralsorcery:itemcoloredlens:6>, 
+	<bloodmagic:blood_rune:9>, <bloodmagic:blood_rune:10>, <bloodmagic:points_upgrade>,
+	<bloodmagic:slate:4>, #<astralsorcery:itemcelestialcrystal>.withTag({}),
+	#<thaumcraft:phial:1>.withTag({Aspects: [{amount: 10, key: "vitium"}]}), 
+	<thaumcraft:mechanism_complex>]);
+	
+	
+# Blutonium Block // Melodic Block
+//	recipes.remove(<minecraft:dirt>);
```
Comment (NielsPilgaard) [inline scripts/Endergy.zs:91]: errr? :P

File: scripts/Endergy.zs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	recipes.addShapedMirrored("Melodic Block", 
+	<enderio:block_alloy_endergy:2>, 
+	[[<ore:blockMithril>, <ore:blockCobalt>, <ore:blockMithril>],
+	[<ic2:nuclear:7>, <ic2:nuclear:7>, <ic2:nuclear:7>], 
+	[<ore:depletedFuelLEU235>, <actuallyadditions:block_crystal_empowered:1>, <ore:depletedFuelLEU235>]]);
+
+# Ludicrite Block // Stellar Block
+//	recipes.remove(<minecraft:blockmetals:4>);
```
Comment (NielsPilgaard) [inline scripts/Endergy.zs:99]: I think you forgot to remove this.

**Review by NielsPilgaard — APPROVED** [review]
Looks good to me, great work!

File: .gitignore (hunk 1 above)
Comment (Discomanco) [reply .gitignore:108]: Actually not sure when I added that to .gitignore, but apparently I did :P

File: scripts/DraconicEvolution.zs (hunk 2 above)
Comment (Discomanco) [reply scripts/DraconicEvolution.zs:14]: It's what is used for Rune of Sacrifice, and assuming the reason that one didn't break. I did test it out with freshly made swords and had no issue :D

File: scripts/Endergy.zs (hunk 3 above)
Comment (Discomanco) [reply scripts/Endergy.zs:91]: Yeah, still need tidbits of code cleanup :P

File: scripts/Endergy.zs (hunk 4 above)
Comment (Discomanco) [reply scripts/Endergy.zs:99]: Polishing missing indeed :P

**Comment by Discomanco** [comment]
Only thing left now is the Dark Steel Armor Upgrade recycling, but it has proven difficult to get around

**Comment by NielsPilgaard** [comment]
> Only thing left now is the Dark Steel Armor Upgrade recycling, but it has proven difficult to get around

What seems to be going on there?

**Comment by Discomanco** [comment]
> > Only thing left now is the Dark Steel Armor Upgrade recycling, but it has proven difficult to get around
> 
> What seems to be going on there?

I can't do a wildcard NBT `.anyDamage()`/`.withTag({})` with the machine (or Mek machines), and it also doesn't seem to want to accept an array of exact NBT tags of `val upgradesArray as IData = { "enderio:dsu":"enderio:<upgradeName>", ..` in a `for each/key/value/(key+value) in upgradesArray {`

**Comment by NielsPilgaard** [comment]
> And I really don't feel like doing 27 different recipes for each individual recipe (double that if activated upgrades are to be included too)

Understandable! Would you prefer leaving them out or making the 27 recipes? 😄

**Comment by Discomanco** [comment]
Going to double check with Mek machines, although hopes are low. And if not, I might use a tool for it. I kinda want it in :P

File: scripts/ThermalExpansion.zs [hunk 5]
```diff
@@ -84,7 +84,7 @@ for slab in <ore:slabWood>.items {
 	100000000, 1000000, 
 	<thermalexpansion:frame:132>, 
 	[<gendustry:genetics_processor>, <extrautils2:ingredients:5>, <ic2:plate:16>,
-	<ic2:lapotron_crystal>.anyDamage(), <forestry:thermionic_tubes:12>, <forestry:crafting_material:1>,
+	<ic2:lapotron_crystal:*>, <forestry:thermionic_tubes:12>, <forestry:crafting_material:1>,
```
Comment (NielsPilgaard) [inline scripts/ThermalExpansion.zs:87]: DOES THIS WORK!? Could've sworn I tried that :P

**Review by NielsPilgaard — APPROVED** [review]

**Comment by NielsPilgaard** [comment]
Great work Disco :D

File: scripts/ThermalExpansion.zs (hunk 5 above)
Comment (Discomanco) [reply scripts/ThermalExpansion.zs:87]: It's what's used for Quantum armor, and also why they didn't break :D

**Comment by Shijikori** [comment]
out of curiosity, because I returned to Enigmatica modpacks recently and noticed the removal of Animania... why was it removed? Performance/Stablity reasons?

**Comment by NielsPilgaard** [comment]
Performance yeah 👍

**Pull request merged**